src/run/dependency/dependency-diagram.ts

```diff
@@ -0,0 +1,107 @@
+import { writeFileSync } from "fs";
+import AbstractDependencyGraph, { Logger, ServiceFinder, ServiceGitDescriptionAndOwner } from "./AbstractDependencyGraph.js";
+import DependencyNode from "../../model/DependencyNode.js";
+import { graphviz } from "node-graphviz";
+import open from "open";
+import Service from "../../model/Service.js";
+import { Inventory } from "../../state/inventory.js";
+
+type TraveseType = "service" | "system"
+
+export default class DependencyDiagram extends AbstractDependencyGraph {
+
+    private logger: Logger;
+
+    constructor (logger: Logger, serviceFinder?: ServiceFinder) {
+        super(serviceFinder);
+        this.logger = logger;
+    }
+
+    private generateDotHeader (diagramName: string, options: { rankdir?: string, ranksep?: number, splines?: string, nodesep?: number, fontsize?: number, height?: number, width?: number } = {}): string {
+        let dot = `digraph ${diagramName} {\n`;
```

Comment:
I'd extract the composition of the digraph to a separate builder which this calls and there is a lot of logic here and may help to reduce the logic?